### PR TITLE
virtualization: merge pxe boot process of numburg and beijing

### PR DESCRIPTION
**Pull request by alice-suse**
Changes:
merge pxe boot process of numburg and beijing; 
simplify and reuse code for reboot_and_login process;
rebase the virt_utils package

The new pxe boot code is verified in beijing, and it works. It should be able to work in numburg too, but still need to verify locally.

### CODE REVIEW

**Comment by alice-suse** [comment]
@coolo  pxe fix in numburg and merge the pxe boot code for beijing and numburg is done

File: tests/boot/boot_from_pxe.pm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            $image_name = get_var("REPO_0");
+        }
+
+        my $arch = get_var('ARCH');
+        my $path = "/mnt/openqa/repo/${image_name}/boot/${arch}/loader";
+        my $repo = get_var('HOST') . "/assets/repo/${image_name}";
+        $image_path = "$path/linux initrd=$path/initrd install=$repo";
+    }
+
+    my $type_speed = 20;
```
Comment (okurz) [inline tests/boot/boot_from_pxe.pm]: There is a function `type_string_slow` in lib/utils.pm. Would this work? If you have a good reason why 20 is better this is also ok.

**Comment by okurz** [comment]
see inline comment. And also it would be more correct to write "Nuremberg" or german way "Nürnberg" :-)

Anyway, it is a good improvement like it is so if you can not change these little issues right now we could also merge as is.

File: tests/boot/boot_from_pxe.pm (hunk 1 above)
Comment (alice-suse) [inline tests/boot/boot_from_pxe.pm]: This is the original way in qa_net.pm. I did not change it.

**Comment by alice-suse** [comment]
Will correct to Nuremberg.

**Pull request merged**